src/components/data/data-utils.js

```diff
@@ -1185,13 +1187,26 @@ export const ModalToolButton = ({ icon, text, disabled, ...props }) => {
   ])
 }
 
-export const HeaderOptions = ({ sort, field, onSort, extraActions, children }) => {
-  const columnMenu = h(MenuTrigger, {
+export const HeaderOptions = ({ sort, field, onSort, extraActions, renderSearch, searchByColumn, children }) => {
+  const popup = useRef()
+  const columnMenu = h(PopupTrigger, {
+    ref: popup,
     closeOnClick: true,
     side: 'bottom',
     content: h(Fragment, [
       h(MenuButton, { onClick: () => onSort({ field, direction: 'asc' }) }, ['Sort Ascending']),
       h(MenuButton, { onClick: () => onSort({ field, direction: 'desc' }) }, ['Sort Descending']),
+      renderSearch && h(ConfirmedSearchInput, {
+        'aria-label': 'Filter by column',
+        placeholder: 'Exact match search in column',
```

Comment:
nit: "filter" or "search"? I like "Exact match filter" better than "Exact match search" but either way the `aria-label` and the `placeholder` should be consistent